modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java

```diff
@@ -0,0 +1,546 @@
+import static org.apache.ignite.console.migration.MigrateUtils.getDocument;
+import static org.apache.ignite.console.migration.MigrateUtils.getInteger;
+import static org.apache.ignite.console.migration.MigrateUtils.mongoIdsToNewIds;
+import static org.apache.ignite.console.migration.MigrateUtils.mongoToJson;
+import static org.apache.ignite.console.migration.MigrateUtils.off;
+import static org.apache.ignite.console.utils.Utils.fromJson;
+
+/**
+ * Service to migrate user data from MongoDB to GridGain persistence.
+ */
+@Service
+public class MigrationFromMongo {
+    /** */
+    private static final Logger log = LoggerFactory.getLogger(MigrationFromMongo.class);
+
+    /** */
+    private final TransactionManager txMgr;
+
+    /** */
+    private final AccountsRepository accRepo;
+
+    /** */
+    private final NotebooksRepository notebooksRepo;
+
+    /** */
+    private final ConfigurationsRepository cfgsRepo;
+
+    /** */
+    @Qualifier("migration")
+    private final MigrateActivitiesRepository activitiesRepo;
+
+    /** */
+    protected MongoDatabase mongoDb;
+
+    /** */
+    @Value("${migration.enabled:false}")
+    private boolean enabled;
+
+    /** */
+    @Value("${migration.mongo:console}")
+    private String mongoDbName;
+
+    /**
+     * Initialize migration service.
+     *
+     * @param ignite Ignite.
+     * @param txMgr Transaction manager.
+     * @param accRepo Repository to work with accounts.
+     * @param notebooksRepo Repository to work with notebooks.
+     * @param cfgsRepo Repository to work with configurations.
+     * @param activitiesRepo Repository to work with activities.
+     */
+    public MigrationFromMongo(
+        Ignite ignite,
+        TransactionManager txMgr,
+        AccountsRepository accRepo,
+        NotebooksRepository notebooksRepo,
+        ConfigurationsRepository cfgsRepo,
+        MigrateActivitiesRepository activitiesRepo
+    ) {
+        this.txMgr = txMgr;
+        this.accRepo = accRepo;
+        this.notebooksRepo = notebooksRepo;
+        this.cfgsRepo = cfgsRepo;
+        this.activitiesRepo = activitiesRepo;
+    }
+
+    /**
+     * Migrate from Mongo to GridGain.
+     */
+    public void migrate() {
+        if (!enabled) {
+            log.info("Migration disabled.");
+
+            return;
+        }
+
+        boolean migrationNeeded;
+
+        try(Transaction tx = txMgr.txStart()) {
+            migrationNeeded = accRepo.ensureFirstUser();
+
+            tx.rollback();
+        }
+
+        if (!migrationNeeded) {
+            log.warn("Database was already migrated. Consider to disable migration in application settings.");
+
+            return;
+        }
+
+        log.info("Migration started...");
+
+        MongoClient mongoClient = MongoClients.create();
+
+        try {
+            mongoDb = mongoClient.getDatabase(mongoDbName);
+
+            migrateAccounts();
+
+            migrateEx();
```

Comment:
This method is not needed, override `migrate`

Reply:
Fixed.